specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/AuthConfigs.json

```diff
@@ -0,0 +1,1049 @@
+      "type": "object",
+      "properties": {
+        "routes": {
+          "$ref": "#/definitions/LoginRoutes",
+          "description": "The routes that specify the endpoints used for login and logout requests."
+        },
+        "tokenStore": {
+          "$ref": "#/definitions/TokenStore",
+          "description": "The configuration settings of the token store."
+        },
+        "preserveUrlFragmentsForLogins": {
+          "description": "<code>true</code> if the fragments from the request are preserved after the login request is made; otherwise, <code>false</code>.",
+          "type": "boolean"
+        },
+        "allowedExternalRedirectUrls": {
+          "description": "External URLs that can be redirected to as part of logging in or logging out of the app. Note that the query string part of the URL is ignored.\nThis is an advanced setting typically only needed by Windows Store application backends.\nNote that URLs within the current domain are always implicitly allowed.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "cookieExpiration": {
+          "$ref": "#/definitions/CookieExpiration",
+          "description": "The configuration settings of the session cookie's expiration."
+        },
+        "nonce": {
+          "$ref": "#/definitions/Nonce",
+          "description": "The configuration settings of the nonce used in the login flow."
+        }
+      }
+    },
+    "LoginRoutes": {
+      "description": "The routes that specify the endpoints used for login and logout requests.",
+      "type": "object",
+      "properties": {
+        "logoutEndpoint": {
+          "description": "The endpoint at which a logout request should be made.",
+          "type": "string"
+        }
+      }
+    },
+    "TokenStore": {
+      "description": "The configuration settings of the token store.",
+      "type": "object",
+      "properties": {
+        "enabled": {
+          "description": "<code>true</code> to durably store platform-specific security tokens that are obtained during login flows; otherwise, <code>false</code>.\n The default is <code>false</code>.",
+          "type": "boolean"
+        },
+        "tokenRefreshExtensionHours": {
+          "format": "double",
+          "description": "The number of hours after session token expiration that a session token can be used to\ncall the token refresh API. The default is 72 hours.",
+          "type": "number"
+        },
+        "azureBlobStorage": {
+          "$ref": "#/definitions/BlobStorageTokenStore",
+          "description": "The configuration settings of the storage of the tokens if blob storage is used."
+        }
+      }
+    },
+    "BlobStorageTokenStore": {
+      "description": "The configuration settings of the storage of the tokens if blob storage is used.",
+      "type": "object",
+      "properties": {
+        "sasUrlSettingName": {
```

Comment:
>sasUrlSettingName [](http://example.com/codeflow?start=9&length=17)

is this the actaul sas? if it is a secret you should add the x-ms-secret to it. I did not find it in the example files.

Reply:
sasUrlSettingName is a property which already exists in previous api version as you can see in https://github.com/Azure/azure-rest-api-specs/blob/main/specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/AuthConfigs.json#L481. We do not want to update it in new api version.